### PR TITLE
ce/Improve request performance SC-2980

### PR DESCRIPTION
**Pull request by CeEv**
# Checkliste

- Bis die komplette Checkliste abgearbeitet ist muss das PR-Label WIP gesetzt sein

- Ticket: https://ticketsystem.schul-cloud.org/browse/SC-2980
- Ticket: https://ticketsystem.schul-cloud.org/browse/SC-2981
- Ticket: https://ticketsystem.schul-cloud.org/browse/SC-2983
- Ticket: https://ticketsystem.schul-cloud.org/browse/SC-2943

- [x] WIP PR-Label entfernt, wenn die Checkliste abgearbeitet wurde

## Mehr
Weitere Informationen zur DoD [hier im Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by CeEv** [comment]
Tobi das splitten ist das Ziel dieses PRs.

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- apiEditor.js  6
         

Complexity decreasing per file
==============================
+ api.js  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- apiEditor.js  6
         

Complexity decreasing per file
==============================
+ api.js  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- helpers/apiHelper.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- helpers/apiHelper.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- helpers/apiHelper.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- helpers/apiHelper.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- helpers/apiHelper.js  6
- app.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

File: app.js [hunk 1]
```diff
@@ -195,6 +204,12 @@ app.use((err, req, res, next) => {
 	} else {
 		res.locals.message = err.message;
 	}
+
+	if (err.message.includes('ESOCKETTIMEDOUT')) {
```
Comment (tofixx) [inline app.js]:
```suggestion
	if (res.locals && res.locals.message.includes('ESOCKETTIMEDOUT')) {
```

File: app.js [hunk 2]
```diff
@@ -146,10 +155,10 @@ app.use(async (req, res, next) => {
 	res.locals.notification = req.session.notification;
 	res.locals.inline = req.query.inline || false;
 	setTheme(res);
-	res.locals.domain = process.env.SC_DOMAIN || 'localhost';
+	res.locals.domain = SC_DOMAIN;
 	res.locals.production = req.app.get('env') === 'production';
-	res.locals.env = req.app.get('env') || false;
-	res.locals.SENTRY_DSN = process.env.SENTRY_DSN || false;
+	res.locals.env = req.app.get('env') || false; // TODO: ist das false hier nicht quatsch?
```
Comment (tofixx) [inline app.js]: jap, daher NODE_ENV verwenden und in den globals process.env.NODE_ENV || 'development' setzen?
--> https://stackoverflow.com/a/34228447

File: app.js [hunk 3]
```diff
@@ -195,6 +204,12 @@ app.use((err, req, res, next) => {
 	} else {
 		res.locals.message = err.message;
 	}
+
+	if (res.locals && res.locals.message.includes('ESOCKETTIMEDOUT') && err.options) {
+		const message = `ESOCKETTIMEDOUT by route: ${err.options.baseUrl + err.options.uri}`;
```
Comment (tofixx) [inline app.js]: here we don't know if err.options exist

Comment (CeEv) [reply app.js]: Nur wenn err.options existiert darf er in die Bedingung gehen. Das sollte bei dem Fehler der Fall sein. Wenn nicht soll er nichts machen.

File: app.js (hunk 2 above)
Comment (CeEv) [reply app.js]: req.app.get('env') || false vs process.env.NODE_ENV || 'development' ist nicht unbedingt das gleiche. Bis auf den stackoverflow post habe ich noch nichts zu default values bei get('env') gelesen. 
Zum nachlesen fehlt mir die Zeit und ohne nachlesen ist es zu kritisch zum ändern.

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app.js  3
- helpers/apiHelper.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/schul-cloud/schulcloud-client/pullRequest?prid=4657966&bid=15598568)

**Pull request merged**